File: Assets/Scripts/Objective.js
var objective = "Find 50 Mushrooms!";
var objectiveComplete = "Found 50 Mushrooms!";
var objectiveCompleteBool : boolean = false;
var crystals : int = 0;

function OnTriggerEnter (crystal : Collider) {
	if(crystal.tag=="Player") {
		Debug.Log("Found 1 mushroom!");
		crystals++;	
	}
}
function Update() {
	if(crystals>=50) {
		objectiveCompleteBool = true;
	}
}

function OnGUI() {
	if(objectiveCompleteBool) {
		GUI.Box(Rect(200,250,175,125),"Bjorn");
		GUI.Label(Rect(200,275,125,125),"So now we can make mushrooms stew!");
		if(GUI.Button(Rect(200,375,125,25),"Make Stew?")) {
			Application.LoadLevel("Kitchen");
		}
	}
}

File: Assets/MainScene/Scripts/Objective.cs
using UnityEngine;
using System.Collections;

public class Objective : MonoBehaviour {
	public bool objectiveCompleted;
	public bool task;
	public float time;
	
	void Start() {
		task = true;
		objectiveCompleted = false;
		time = Time.time;
	}
	
	void Update() {
		if(Time.time - time > 4f)
			task = false;
	}
	
	void OnTriggerEnter(Collider player) {
		if(player.tag == "Player") {
			Time.timeScale = 0;
			objectiveCompleted = true;
			renderer.enabled = false;
		}
	}
	
	void OnGUI() {
		if(objectiveCompleted) {
			GUI.Box (new Rect(Screen.width/2 - 100,Screen.height/2 - 100, 200, 50),"");
			GUI.Label (new Rect(Screen.width/2 - 95,Screen.height/2 - 90, 190, 40),"Поздравляем, вы прошли игру!");
			if(GUI.Button(new Rect(Screen.width/2 - 95,Screen.height/2, 190, 40),"В меню")) {
				Application.LoadLevel("MainMenu");
			}
		}
		if(task) {
			GUI.Box (new Rect(Screen.width/2 - 100,Screen.height/2 - 25, 200, 80),"");
			GUI.Label (new Rect(Screen.width/2 - 80,Screen.height/2 - 15, 100, 70),"Добраться до энергетического кристалла");
		}
		
			
	}
}
